PRODDOCS-649: CI/CD API (#9083)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/continuous-integration.md

The [<Constant name="dbt" /> scheduler](/docs/deploy/job-scheduler) executes CI jobs differently from other deployment jobs in these important ways:

- [**Concurrent CI checks**](#concurrent-ci-checks) &mdash; The scheduler can execute multiple CI runs from the same <Constant name="dbt" /> CI job concurrently (in parallel) when appropriate.
- [**Smart cancellation of stale builds**](#smart-cancellation-of-stale-builds) &mdash; The scheduler cancels stale in-flight CI runs when you push new commits to the PR.
- [**Run slot treatment**](#run-slot-treatment) &mdash; CI runs don't consume a run slot.
- [**SQL linting**](#sql-linting) &mdash; When enabled, linting runs on all SQL files in your project as a step before your CI job builds.

### Concurrent CI checks <Lifecycle status="self_service,managed,managed_plus" />

> When teammates collaborate on the same dbt project and open pull requests in the same dbt repository, the same CI job can run for each qualifying event.

Reviewer: ```suggestion
When you collaborate on a dbt project with your team and open pull requests in the same dbt repository, the same CI job can run for each qualifying event.
```

Each CI run writes to a dedicated, temporary schema that is tied to its pull request. That isolation lets <Constant name="dbt" /> execute CI runs _concurrently_ instead of _sequentially_. This differs from deployment <Constant name="dbt" /> jobs.

> No teammate has to wait for someone else’s CI run to finish before their own check can start. Concurrent CI checks help your whole team test and integrate dbt code faster.

Reviewer: ```suggestion
You don't have to wait for someone else’s CI run to finish before your own check can start. Concurrent CI checks help your whole team test and integrate dbt code faster.
```

Available on [<Constant name="dbt" /> release tracks](/docs/dbt-versions/cloud-release-tracks) and <Constant name="dbt" /> Starter or Enterprise-tier accounts.

When [enabled for your CI job](/docs/deploy/ci-jobs#set-up-ci-jobs), dbt invokes [SQLFluff](https://sqlfluff.com/). SQLFluff is a modular, configurable SQL linter. It warns you about complex functions, syntax, formatting, and compilation errors.

> SQLFluff linting is not yet supported for <Constant name="dbt_platform" /> jobs that run on the <Constant name="fusion_engine" />. For more information, refer to [<Constant name="fusion" /> limitations](/docs/fusion/supported-features#limitations).

Reviewer: This might be good as a callout
